### PR TITLE
…

### PR DESCRIPTION
**Pull request by nnunn**
fixing printing bug, adding quotes around subject searches, fixing author searches.

### CODE REVIEW

**Comment by pgwillia** [comment]
I'm about to take a look at this.  Just wondering what was the problem and expected behaviour with the "printing bug?"  Similar questions about subject and author searches; What was the problem and expected behaviour?

**Comment by nnunn** [comment]
The full record was not printing in Chrome. This was fixed by removing the bootstrap class "hidden-xs". See issue #1115 

The subject search within a full record needed to be in quotes so that the exact string was searched. see issue #1109 

The author search fixes issue #938  - the author link within a record should return the same results as an advanced search of that author.

Hope that helps - sorry I wasn't clearer in the initial pull request.

**Comment by nnunn** [comment]
If you print a record view from Chrome (for example https://library.ualberta.ca/catalog/3053737) you only get the table holdings and not the metadata for that record. This only happens in chrome and it appears it is because chrome is hiding divs with the bootstrap class "hidden-xs" - since we actually don't need the record metadata hidden on small devices, I removed this class. This now prints the full record metadata in Chrome.

I can move the string manipulations into a helper - are you specifically talking about the subject string manipulation?

I don't actually know how to set up regression tests...

File: app/views/catalog/_subjects.html.erb [hunk 1]
```diff
@@ -8,13 +8,13 @@
             <% i = 0 %>
             <% sbstring = "" %>
```
[the remark this answers is not in the document]
Comment (nnunn) [reply app/views/catalog/_subjects.html.erb]: sure that makes sense. I'll do that and push it up.

**Comment by nnunn** [comment]
That would be great - thanks! Maybe after you set it up you could walk me through it so I know how to do it next time?

**Comment by nnunn** [comment]
Ok, I've created and pushed the subject helper. Let me know if it looks ok.

Do you have any further questions about the printing bug?

**Comment by pgwillia** [comment]
@nnunn Sorry I didn't get back to this today.  If you look at the "All checks have failed" you can see that Rubocop has some feedback on the code style in the helper.  A quick way to resolve most of those issues is to run `rubocop -a` which auto-corrects the nags when possible.

Still can't reproduce the chrome bug, but the change seems logical to me.

I'll look at writing the test tomorrow. The ability to test is improved by https://github.com/ualbertalib/discovery/pull/1227, but it there is a change to how the application is configured that would have to be matched by Neil in ansible.

File: app/helpers/subject_helper.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module SubjectHelper
+     def subject_query(subject, index)
```
Comment (pgwillia) [inline app/helpers/subject_helper.rb]: Great!  Thanks this makes it easier to read, test and talk about this code.

I think there are some room for improvements here.  First at `app/views/catalog/_subjects.html.erb:7` by passing `subject_query(subject.take(index+1))` we can change the method to `def subject_query(subjects)` and not have to use index or a counter to reason about the arrays of strings.

Second, we can use `subjects = subjects.map {|subject| subject.delete('.&').tr('+', ' ')}` to clean up the string.

Third, we can use `subjects.join.strip` to replace `sbstring +=`

Finally, I think making this a phrase replaces the need to do some of the string cleanup. I did some digging and the & is from [here](https://github.com/ualbertalib/discovery/issues/910), but I couldn't see where . and + were important. Do you know? Is there a good example?
If I'm right then this method becomes
```
def subject_query(subjects)
  "\"#{subjects.join.strip}\""
end
```
Here's an example of the test that I've written for this helper.
```
# spec/helpers/search_helper_spec.rb
require "spec_helper"

describe SubjectHelper do
  describe "#subject_query" do
    it "should handle ampersands, periods and plus" do
      # https://github.com/ualbertalib/discovery/issues/910
      expect(helper.subject_query(['Library & Information Studies'])).to eq "\"Library  Information Studies\""

      expect(helper.subject_query(['England and Wales. Parliament'])).to eq "\"England and Wales Parliament\""
      expect(helper.subject_query(['C++ (Computer program language)'])).to eq "\"C   (Computer program language)\""
    end
    
    # https://github.com/ualbertalib/discovery/issues/1109
    it "should handle subject chaining" do
      expect(helper.subject_query(['Michif language ', ' Juvenile sound recordings'])).to eq "\"Michif language  Juvenile sound recordings\""
    end
  end
end
```

Comment (nnunn) [reply app/helpers/subject_helper.rb]: Great - that is a lot cleaner. Thanks!

So if I were to create a test like that how would I actually run it?

Comment (nnunn) [reply app/helpers/subject_helper.rb]: I don't think the + and . break anything, but the query strings look a litte odd with them in there. I think from a user perspective it makes sense to strip them out.

Comment (pgwillia) [reply app/helpers/subject_helper.rb]: I've added these changes as a commit to this branch.  If you `git pull` you should have the test.  You can run it and all the tests `bundle exec rake spec`

**Comment by nnunn** [comment]
Are there any other changes I should make on this pr?

File: app/helpers/subject_helper.rb (hunk 2 above)
Comment (nnunn) [reply app/helpers/subject_helper.rb]: awesome thanks!

**Pull request merged**